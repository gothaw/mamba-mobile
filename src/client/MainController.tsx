import { createContext } from "react";
import { Route, NativeRouter, Routes } from "react-router-native";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

const MainController = () => {
  const Context = createContext({});

  return (
    <Context.Provider value={{}}>
      <NativeRouter>
        <Routes>
          <Route path="/" element={<MainMenu/>}/>
          <Route path="/game" element={<Game/>}/>
        </Routes>
      </NativeRouter>
    </Context.Provider>
  );
}

export default MainController;