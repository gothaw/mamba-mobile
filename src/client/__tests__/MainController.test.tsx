import { render, waitFor } from "@testing-library/react-native";

import MainController from "../MainController";

describe("MainController", () => {
  let component;

  beforeEach(() => {
    component = render(<MainController/>);
  });

  it("should render without crashing", async () => {
    await waitFor(() => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  it("should not render if not ready", () => {
    expect(component.toJSON()).toBe(null);
  });
});