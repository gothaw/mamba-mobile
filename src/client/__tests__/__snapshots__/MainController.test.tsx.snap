// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainController should render without crashing 1`] = `null`;
